apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/QueryUtils.java

```diff
@@ -69,62 +76,54 @@ public static Predicate createLastUpdatedPredicateInstant(
-        range.getUpperBoundAsInstant() == null ? null : range.getUpperBoundAsInstant().toInstant();
-    Predicate lowerBoundPredicate;
-    Predicate upperBoundPredicate;
+  public static Predicate createDateRangePredicate(
+      CriteriaBuilder builder, DateRangeParam dateRange, Expression<LocalDate> dateExpression) {
+    final List<Predicate> predicates = new ArrayList<>();
 
+    final DateParam lowerBound = dateRange.getLowerBound();
     if (lowerBound != null) {
-      switch (range.getLowerBound().getPrefix()) {
-        case GREATERTHAN_OR_EQUALS:
-          lowerBoundPredicate = cb.greaterThanOrEqualTo(lastUpdatedPath, lowerBound);
-          break;
-        case GREATERTHAN:
-          lowerBoundPredicate = cb.greaterThan(lastUpdatedPath, lowerBound);
-          break;
-        default:
-          throw new IllegalArgumentException("_lastUpdate lower bound has an invalid prefix");
+      final LocalDate from =
+          lowerBound.getValue().toInstant().atOffset(ZoneOffset.UTC).toLocalDate();
+
+      if (ParamPrefixEnum.GREATERTHAN.equals(lowerBound.getPrefix())) {
+        predicates.add(builder.greaterThan(dateExpression, from));
+      } else if (ParamPrefixEnum.GREATERTHAN_OR_EQUALS.equals(lowerBound.getPrefix())) {
+        predicates.add(builder.greaterThanOrEqualTo(dateExpression, from));
+      } else {
+        throw new IllegalArgumentException(
+            String.format("Unsupported prefix supplied %s", lowerBound.getPrefix()));
       }
-    } else {
-      lowerBoundPredicate = null;
     }
 
+    final DateParam upperBound = dateRange.getUpperBound();
     if (upperBound != null) {
-      switch (range.getUpperBound().getPrefix()) {
-        case LESSTHAN:
-          upperBoundPredicate = cb.lessThan(lastUpdatedPath, upperBound);
-          break;
-        case LESSTHAN_OR_EQUALS:
-          upperBoundPredicate = cb.lessThanOrEqualTo(lastUpdatedPath, upperBound);
-          break;
-        default:
-          throw new IllegalArgumentException("_lastUpdate upper bound has an invalid prefix");
-      }
-      if (lowerBoundPredicate == null) {
-        return cb.or(cb.isNull(lastUpdatedPath), upperBoundPredicate);
+      final LocalDate to = upperBound.getValue().toInstant().atOffset(ZoneOffset.UTC).toLocalDate();
+
+      if (ParamPrefixEnum.LESSTHAN_OR_EQUALS.equals(upperBound.getPrefix())) {
+        predicates.add(builder.lessThanOrEqualTo(dateExpression, to));
+      } else if (ParamPrefixEnum.LESSTHAN.equals(upperBound.getPrefix())) {
+        predicates.add(builder.lessThan(dateExpression, to));
       } else {
-        return cb.and(lowerBoundPredicate, upperBoundPredicate);
-      }
-    } else {
-      if (lowerBoundPredicate == null) {
         throw new IllegalArgumentException(
-            ("_lastUpdate upper and lower bound cannot both be null"));
-      } else {
-        return lowerBoundPredicate;
+            String.format("Unsupported prefix supplied %s", upperBound.getPrefix()));
```

Comment:
The other method that git is comparing this code to is not being changed.  It was actually deleted because it was an exact duplicate of another method in this same class.  The logic at the end of `createLastUpdatedPredicate()` to combine the lower and upper bound predicates was somewhat complicated.

The logic in `createDateRangePredicate()` is simpler.  It builds the upper and lower bound predicates separately and adds them to a list.  After both are built (or not built) the list will contain 0-3 predicates.  These are simply passed to the `and()` method to build the result predicate.

We could throw an extra exception here if the list is empty but we check for the existence of a non-empty date range param before calling this.

Reply:
Thanks, i think the diff was just confusing like you mentioned in the engineering sync.